architect/systems/spectrometers.py

```diff
@@ -77,40 +78,38 @@ def get_ratio_cropped_light_through_slit(self):
 
     def get_signal_to_noise(self, radiance: LUT, wavelength):
         """Get the signal to noise ratio of the system.
-
         Ref: https://www.notion.so/utat-ss/Signal-to-Noise-6a3a5b8b744d41ada40410d5251cc8ac
-
         """
 
-        snr = signal / self.get_noise()
+        snr = (
+            self.get_signal(radiance=radiance, wavelength=wavelength) / self.get_noise(wavelength, radiance)
+        )
 
         return snr
 
-    def get_signal(wavelength):
+    def get_signal(self, wavelength, radiance: LUT):
         """Get the signal.
-
         Ref: https://www.notion.so/utat-ss/Signal-1819461a3a2b4fdeab8b9c26133ff8e2
-
         """
         assert self.sensor is not None, "A sensor component must be specified."
         assert self.foreoptic is not None, "A foreoptic component must be specified."
         assert self.slit is not None, "A slit component must be specified."
 
         signal1 = (
-            (wavelength / (const.h * const.c))
+            (wavelength.to(unit.m) / (const.h * const.c))
             * radiance(wavelength)
             * (math.pi / 4)
-            * self.sensor.get_integration_time()
+            * (self.sensor.get_integration_time().to(unit.s))
         )  # [sr-1 m-3]
         assert signal1.unit == unit.sr**-1 * unit.m**-3
-        signal2 = self.sensor.get_pixel_area() / (
+        signal2 = self.sensor.get_pixel_area().to(unit.m**2) / (
             ((self.foreoptic.get_f_number()).decompose()) ** 2 * 1 / unit.sr
         )  # [m2 sr]
         assert signal2.unit == unit.m**2 * unit.sr
         signal3 = (
             (self.sensor.get_efficiency(wavelength)).decompose()
             * unit.electron
-            * self.get_transmittance()
+            * self.get_transmittance().value
```

Comment:
Is there a reason we do unit conversions here? Astropy should be able to handle the units automatically

Reply:
Yeah, that's what I thought too, but when I tried to run it without these units, it kept giving me assertion errors. I added the units, and it seemed to work, so I'm not quite sure.

Reply:
Ok, we can leave it as is then. I'll look into this as part of my task on refactoring the SNR function